### PR TITLE
Fixed held items going invisible on inventory hide after being swapped

### LINKED ISSUE #6781: Visual but when swapping worn items then hiding inventory
### Labels

[BUG]

### Description

Swapping worn items via toggle button then hiding inventory hides the item that when to your hand

### Steps to reproduce

1. Swap worn items by having a wearable that shares a type with an occupied inventory slot in the active hand and use the toggle inventory button/hotkey.
2. Switch to other hand and hide inventory HUD using toggle inventory. The swapped worn item will be hidden from the hand slot along with the inventory HUD and can't be clicked while in this state, but otherwise functions as expected. Icon becomes visible again when HUD toggled back on or item changes location.

### Expected Behavior

Item in hand slot should remain visible

### Additional Information & Screenshots

image
image

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
Fixes #6781
[BUG]
## About the PR 
The only thing this actually does is add `inventory_items -= C;` to the macro, but to do that I had to be able to actually read it so it's multiline now.

## Why's this needed? 
Fixes minor visual glitch.

### CODE REVIEW

**Comment by TobleroneSwordfish** [comment]
Also that file has so much code duplication that I will leave to someone who understands the code better to fix

File: code/datums/hud/human.dm [hunk 1]
```diff
@@ -310,7 +310,26 @@
...
+					#define autoequip_slot(slot, var_name)\
+						if (master.can_equip(I, master.slot) && !istype(I.loc, /obj/item/parts) && !(master.var_name && master.var_name.cant_self_remove))\
+						{\
+							master.u_equip(I);\
+							var/obj/item/C = master.var_name;\
+							if (C)\
+							{\
+								/*master.u_equip(C);*/\
+								C.unequipped(master);\
+								inventory_items -= C;\
```
Comment (ZeWaka) [inline code/datums/hud/human.dm:323]: this line is the change btw

Comment (ZeWaka) [reply code/datums/hud/human.dm:323]: why are we not using the `remove_item(x)` wrapper?

Comment (TobleroneSwordfish) [reply code/datums/hud/human.dm:323]: good question, this PR is just a fix to one very specific bug

Comment (ZeWaka) [reply code/datums/hud/human.dm:323]: I think we should be using it here

**Pull request merged**